vdr/vdr.go

```diff
@@ -127,7 +133,22 @@ func (r VDR) Resolve(id did.DID, metadata *types.ResolveMetadata) (*did.Document
 
 // Update updates a DID Document based on the DID and current hash
 func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, metadata *types.DocumentMetadata) error {
-	return r.store.Update(id, current, next, metadata)
+	resolverMetada := &types.ResolveMetadata{
+		Hash:             &current,
+		AllowDeactivated: false,
+	}
+	_, meta, err := r.store.Resolve(id, resolverMetada)
+	if err != nil {
+		return err
+	}
+	payload, err := json.Marshal(next)
+	if err != nil {
+		return err
+	}
+	// TODO: look into the controller of the did for a signing key
+	keyID := next.Authentication[0].ID.String()
```

Comment:
I think it must be signed by a key that was present in the _previous_ (current) version of the document, so from the `Resolve` action